lib/srv/db/cloud/iam.go

```diff
@@ -61,31 +91,57 @@ func NewIAM(ctx context.Context, config IAMConfig) (*IAM, error) {
 		return nil, trace.Wrap(err)
 	}
 	return &IAM{
-		cfg: config,
-		log: logrus.WithField(trace.Component, "iam"),
+		cfg:   config,
+		log:   logrus.WithField(trace.Component, "iam"),
+		tasks: make(chan iamTask, defaultIAMTaskQueueSize),
 	}, nil
 }
 
+// Start starts the IAM configurator service.
+func (c *IAM) Start(ctx context.Context) error {
+	// DELETE IN 11.0.
+	c.migrateInlinePolicy(ctx)
```

Comment:
That is true. Though I think it's usually expected that the access can be "broken" at certain level while the database service is down getting an upgrade. The new policies should be added back within seconds. It's even less an issue in HA mode if services are upgraded at different times (at least one of the old or the new policy should exist).

On the other hand, if we do feel mitigation is necessary, we can either copy the entire old policy to the new policy if new does not exists, or migrate action/resource to new policy one by one. 

What do you think?

Reply:
I think if we can do it in a way that does not prevent access, we should. Esp. since like this would be totally self-inflicted.

What if we don't delete the old policy unconditionally, but only if the new one is already present? This way it will be deleted, say, on the second restart. Or e.g. launch a goroutine that will delete it in 10 minutes or something instead of deleting right upon startup.